File: pipeline/lib/warehouse_paths.py
from __future__ import annotations

import os
from pathlib import Path

ROOT = Path(__file__).resolve().parents[2]
WAREHOUSE_DIR = ROOT / "data-private" / "warehouse"
SPORTS_DB_PATH = WAREHOUSE_DIR / "sports.db"


def _env_path(name: str) -> Path | None:
    value = os.environ.get(name)
    if not value:
        return None
    return Path(value).expanduser()


def warehouse_path_for_sport(sport: str = "shared") -> Path:
    sport_key = "".join(character for character in sport.upper() if character.isalnum() or character == "_")
    return (
        _env_path(f"SLATE_{sport_key}_WAREHOUSE_DB")
        or _env_path("SLATE_WAREHOUSE_DB")
        or SPORTS_DB_PATH
    )


def tennis_warehouse_path() -> Path:
    return warehouse_path_for_sport("tennis")


def mlb_warehouse_path() -> Path:
    return warehouse_path_for_sport("mlb")

File: pipeline/lib/warehouse-paths.mjs
import path from 'node:path'

export const rootDir = path.resolve(import.meta.dirname, '..', '..')
export const warehouseDir = path.join(rootDir, 'data-private', 'warehouse')
export const sportsDbPath = path.join(warehouseDir, 'sports.db')

const envPath = (name) => {
  const value = process.env[name]
  return value ? path.resolve(value) : null
}

export const warehousePathForSport = (sport = 'shared') => {
  const sportKey = String(sport).toUpperCase().replace(/[^A-Z0-9_]+/g, '')
  return envPath(`SLATE_${sportKey}_WAREHOUSE_DB`) || envPath('SLATE_WAREHOUSE_DB') || sportsDbPath
}

export const tennisWarehousePath = () => warehousePathForSport('tennis')
export const mlbWarehousePath = () => warehousePathForSport('mlb')
